sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/StartTransferUploadTests.cs

```diff
@@ -0,0 +1,677 @@
+            await using DisposingShare testShare = await GetTestShareAsync();
+
+            List<DataTransferOptions> optionsList = new List<DataTransferOptions>() { options };
+            await UploadShareFilesAndVerify(
+                share: testShare.Share,
+                blobCount: optionsList.Count,
+                options: optionsList);
+
+            // Assert
+            Assert.IsTrue(progressSeen);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFileSize_SmallChunk()
+        {
+            long fileSize = Constants.KB;
+            int waitTimeInSec = 25;
+            DataTransferOptions options = new DataTransferOptions()
+            {
+                InitialTransferSize = 100,
+                MaximumTransferChunkSize = 200,
+            };
+
+            // Arrange
+            await using DisposingShare testShare = await GetTestShareAsync();
+            List<DataTransferOptions> optionsList = new List<DataTransferOptions>() { options };
+
+            await UploadShareFilesAndVerify(
+                size: fileSize,
+                waitTimeInSec: waitTimeInSec,
+                share: testShare.Share,
+                options: optionsList);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFile_Overwrite_Exists()
+        {
+            // Arrange
+            // Create source local file for checking, and source blob
+            await using DisposingShare testShare = await GetTestShareAsync();
+            using DisposingLocalDirectory testDirectory = DisposingLocalDirectory.GetTestDirectory();
+            string shareFileName = GetNewFileName();
+            string localSourceFile = Path.Combine(testDirectory.DirectoryPath, shareFileName);
+            int size = Constants.KB;
+            int waitTimeInSec = 10;
+
+            // Create file
+            ShareFileClient destClient = await CreateShareFile(testShare.Share.GetRootDirectoryClient(), localSourceFile, shareFileName, size);
+
+            // Act
+            // Create options bag to overwrite any existing destination.
+            DataTransferOptions options = new DataTransferOptions()
+            {
+                CreationPreference = StorageResourceCreationPreference.OverwriteIfExists,
+            };
+            List<DataTransferOptions> optionsList = new List<DataTransferOptions>() { options };
+            List<string> shareFileNames = new List<string>() { shareFileName };
+
+            // Start transfer and await for completion.
+            await UploadShareFilesAndVerify(
+                share: testShare.Share,
+                size: size,
+                waitTimeInSec: waitTimeInSec,
+                shareFileNames: shareFileNames,
+                options: optionsList);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFile_Overwrite_NotExists()
+        {
+            // Arrange
+            // Create source local file for checking, and source blob
```

Comment:
I've been seeing this comment a lot but the local file creation seems to be handled in `UploadShareFilesAndVerify()`, so it's a bit confusing as to what it means here.

Reply:
yeah I think just copy and pasted the comment without thinking. removed